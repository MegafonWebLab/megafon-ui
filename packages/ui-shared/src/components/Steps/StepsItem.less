@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-steps-item');

@{b} {
    display: flex;

    &__step-number {
        .h5();
        display: flex;
        align-items: center;
        justify-content: center;
        flex-shrink: 0;
        width: 40px;
        height: 40px;
        border-radius: 50%;
        background-color: @green;
        color: @clearWhite;
    }

    &__text-wrapper {
        padding-top: 8px;
        padding-left: 16px;
    }
}
