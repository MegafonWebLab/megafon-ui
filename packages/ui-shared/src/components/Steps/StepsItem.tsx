import React from 'react';
import PropTypes from 'prop-types';
import { cnCreate, Paragraph } from '@megafon/ui-core';
import './StepsItem.less';

interface IStepsItem {
    /** Номер элемента */
    index: number;
    /** Текст */
    text: string;
}

const cn = cnCreate('mfui-beta-steps-item');
const StepsItem: React.FC<IStepsItem> = ({ index, text }) => (
    <div className={cn()}>
        <span className={cn('step-number')}>
            {index}
        </span>
        <div className={cn('text-wrapper')}>
            <Paragraph hasMargin={false}>{text}</Paragraph>
        </div>
    </div>
);

StepsItem.propTypes = {
    index: PropTypes.number.isRequired,
    text: PropTypes.string.isRequired,
};

export default StepsItem;
