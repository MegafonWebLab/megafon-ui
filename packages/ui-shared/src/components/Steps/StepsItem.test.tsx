import React from 'react';
import { shallow } from 'enzyme';
import StepsItem from './StepsItem';

describe('StepsItem', () => {
    it('should render component', () => {
        const wrapper = shallow(
            <StepsItem index={1} text="text" />);

        expect(wrapper).toMatchSnapshot();
    });
});
