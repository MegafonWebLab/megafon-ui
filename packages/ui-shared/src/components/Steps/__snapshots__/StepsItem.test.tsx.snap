// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepsItem should render component 1`] = `
<div
  className="mfui-beta-steps-item"
>
  <span
    className="mfui-beta-steps-item__step-number"
  >
    1
  </span>
  <div
    className="mfui-beta-steps-item__text-wrapper"
  >
    <Paragraph
      hasMargin={false}
    >
      text
    </Paragraph>
  </div>
</div>
`;
